KAFKA-8155: Add 2.1.1 release to system tests

mjsax opened the pull request:
*More detailed description of your change,
if necessary. The PR title and PR message become
the squashed commit message, so use a separate
comment to ping reviewers.*

*Summary of testing strategy (including rationale)
for the feature or bug fix. Unit and/or integration
tests are expected for any behaviour change and
system tests should be considered for larger changes.*

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

mjsax committed:
KAFKA-8155: Add 2.1.1 release to system tests

mjsax commented on streams/upgrade-system-tests-21/src/test/java/org/apache/kafka/streams/tests/StreamsUpgradeTest.java:
```diff
@@ -59,40 +59,33 @@ public static void main(final String[] args) throws Exception {
         final KafkaStreams streams = new KafkaStreams(builder.build(), config);
         streams.start();
 
-        Runtime.getRuntime().addShutdownHook(new Thread() {
```
Java8 cleanup only

mjsax replied:
Similar cleanup https://github.com/apache/kafka/pull/6598

mjsax replied:
Similar cleanup in https://github.com/apache/kafka/pull/6599

mjsax replied:
Different PR for different modules to allow corresponding cherry-picking to older branches.

mjsax commented on streams/upgrade-system-tests-21/src/test/java/org/apache/kafka/streams/tests/StreamsUpgradeTest.java:
```diff
@@ -59,40 +59,33 @@ public static void main(final String[] args) throws Exception {
     }
 
     private static <K, V> ProcessorSupplier<K, V> printProcessorSupplier() {
-        return new ProcessorSupplier<K, V>() {
```
Java8 cleanup only

mjsax commented:
Triggered _all_ system tests: https://jenkins.confluent.io/job/system-test-kafka-branch-builder/2515/

Call for review @guozhangwang @bbejeck @vvcephei @ableegoldman @cadonna

mjsax commented:
This should be cherry-picked to `2.2` branch. Not sure about `2.1` ?

bbejeck commented:
>This should be cherry-picked to 2.2 branch. Not sure about 2.1 ?

I'd say do `2.1` as well as IMHO we want the latest versions on the system tests.

bbejeck commented:
system test failed, but it was during the setup, not an actual test failure.  So I've kicked off another build https://jenkins.confluent.io/job/system-test-kafka-branch-builder/2518/

mjsax commented:
Updated `Dockerfile`. Retrigger system tests: https://jenkins.confluent.io/job/system-test-kafka-branch-builder/2520/

mjsax commented:
System test failed. It did not download `2.1.1` artifact -- not sure why -- maybe I started the build before it could sync on the docker image update. Retriggered:

https://jenkins.confluent.io/job/system-test-kafka-branch-builder/2525/

mjsax commented:
Rebased to resolve merge conflict. Also fixed the artifact problem. Retriggered system tests: https://jenkins.confluent.io/job/system-test-kafka-branch-builder/2590/

vvcephei reviewed (approved):
Thanks!

guozhangwang commented:
LGTM.

guozhangwang commented:
retest this please

mjsax commented:
Currently system tests on `trunk` are broken -- we need to wait for a fix IMHO, before we can merge this. Thoughts?

mjsax commented:
Rebased this to get latest system test fixed. Retriggered system tests: https://jenkins.confluent.io/job/system-test-kafka-branch-builder/2630/

mjsax committed:
Reverted some trouble shooting debug logs

mjsax commented:
Java8 failed (test results are no longer available). Java11 passed.

Retest this please.

mjsax commented:
System test results not available any longer. Retriggered: https://jenkins.confluent.io/job/system-test-kafka-branch-builder/2654/

mjsax commented:
System tests passed.

Java8 failed (test results are no longer available). Java11 passed.

Retest this please.

mjsax commented:
Merged to `trunk` and cherry-picked to `2.3` and `2.2` branches.

ijuma commented:
@mjsax you broke the 2.2 system tests due to a badly resolved conflict:

https://github.com/apache/kafka/commit/c450bfc2911fa121e18815a15b5fd7f087fd03ac#diff-9f5a3c3b9b36f4d6c32b238f5ec80b89R125

After the commit above, a single system test was being executed:

http://confluent-kafka-2-2-system-test-results.s3-us-west-2.amazonaws.com/2019-06-13--001.1560391755--apache--2.2--bda2a02/report.html
